src/ol/interaction/modifyinteraction.js

```diff
@@ -959,6 +1044,12 @@ ol.interaction.Modify.prototype.insertVertex_ = function(segmentData, vertex) {
       coordinates = geometry.getCoordinates();
       coordinates[depth[0]].splice(index + 1, 0, vertex);
       break;
+    case ol.geom.GeometryType.RECTANGLE:
```

Comment:
Okay, but that doesn't change the need for origSegment, right? Because you're still adding the segments to `this.dragSegments_` and using those during dragging.